thrust/detail/execute_with_dependencies.h

```diff
@@ -189,7 +189,7 @@ struct execute_with_allocator_and_dependencies
         return std::move(dependencies);
     }
 
-    typename std::remove_reference<Allocator>::type&
+    typename std::add_lvalue_reference<Allocator>::type
     __host__
```

Comment:
We should probably move `__host__` before the return type.

Reply:
Let's open an issue and do this as a separate commit to avoid blocking this change from landing.